### PR TITLE
[MM-42526] When a profile picture fails to load,  fall back to the user's default picture

### PR DESCRIPTION
**Pull request by kyeongsoosoo**
#### Summary

If user's profile picture fails to load, replace src with default profile src(I used the regex).
`Client4.getUsersRoute() + '/' + userId + '/image?_=' + lastPictureUpdate` 
-> 
`Client4.getUsersRoute() + '/' + userId + '/image/default'` 

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/19815

#### Release Note

```release-note
add fallback src to an avatar
```

### CODE REVIEW

**Review by michelengelen — CHANGES_REQUESTED** [review]

File: components/widgets/users/avatar/avatar.tsx [hunk 1]
```diff
@@ -17,6 +20,9 @@ type Props = {
 
 type Attrs = HTMLAttributes<HTMLElement>;
 
+const isURLForUser = (url: string | undefined): url is string => Boolean(url && url.startsWith(Client4.getUsersRoute()));
```
[the remark this answers is not in the document]
Comment (kyeongsoosoo) [reply components/widgets/users/avatar/avatar.tsx]: @michelengelen Thanks for the tip! I made a mistake😅

**Review by michelengelen — APPROVED** [review]
LGTM... thanks!

**Review by BenCookie95 — APPROVED** [review]
LGTM, nice work!

**Review by furqanmlk — COMMENTED** [review]
Hi @kyeongsoosoo can you please provide me broken image.

**Comment by hmhealey** [comment]
This error case could be caused by a few different things (like the image being deleted on the server or a network issue), but the best way to repro this would probably be to block the request using the browser's dev tools. You can do that by:
1. Right clicking on the user's profile picture and copy the image link. It'll look like `https://mattermost-webapp-pr-10810.test.mattermost.cloud/api/v4/users/u8hqnuam9intmydrg7kbepy6we/image?_=1658954268819` which is the current url for user-1's profile picture on that test server.
2. Go to the Network tab of the browser's dev tools and find how to show blocked requests. On Firefox, it's the 🚫 icon to the right of the filter box.
3. Enter the URL from step 1 to block
4. Refresh the page

Or you can take my word that it works based on these screenshots:
|Normal (request allowed)|Error (request blocked)|
|-|-|
|Screen Shot 2022-07-27 at 4 41 23 PM|Screen Shot 2022-07-27 at 4 41 31 PM|

**Comment by kyeongsoosoo** [comment]
> This error case could be caused by a few different things (like the image being deleted on the server or a network issue), but the best way to repro this would probably be to block the request using the browser's dev tools. You can do that by:
> 
@furqanmlk  I tested it in a similar way!

**Review by furqanmlk — APPROVED** [review]
Thanks @kyeongsoosoo  working fine


**With Profile Picture**
image

**Without Profile Picture**
image

**Comment by hmhealey** [comment]
Great. Thanks again for the PR!

**Pull request merged**